/*Task 6:

Imagine a row of linked boxes, like a chain of nodes in a linked list with length N, with Shyaam holding the first box. 

Each box is represented by a number Ai (0 <= i < N). The challenge Ram presents is to rearrange the order of these linked boxes in reverse, making the last box the new first one. 
Can you help Shyaam conquer this task?

Input Format
The first line contain integer N representing the length of chain.

The second line consist of N spcaed integer representing the elements of chain.

Output Format
Print the chain of boxes after reversing.

Constraints
0 < N <= 10^6
-10^4 <= Ai <= 10^4

Sample Testcase 0
Testcase Input
7
9 10 4 2 7 3 6
Testcase Output
6 3 7 2 4 10 9
Explanation
We are reversing the order of the boxes.

Sample Testcase 1
Testcase Input
5
0 6 -1 3 2
Testcase Output
2 3 -1 6 0
Explanation
We are reversing the order of boxes.*/





import java.util.*;
public class calc {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n= sc.nextInt();
        int arr[]=new int[n];
        for(int i =0;i<=n-1;i++){
            arr[i]=sc.nextInt();
            
        }
        for(int i=n-1;i>=0;i--){
            System.out.println(arr[i]);
        }
    }
}
